build/container/Makefile

```diff
@@ -36,104 +64,238 @@ PROMETHEUS_VERSION             ?= $(call get-nri-version,nri-prometheus)
+	@(cp $(AGENT_BUILD_TARGET_DIR)/$(GOOS)-newrelic-infra_$(GOOS)_$(DOCKER_ARCH)/* $(PROJECT_WORKSPACE)/)
 
 .PHONY : build/base
 build/base : workspace/assets
 build/base : base/get-integrations
 build/base : base/get-infra-agent
 build/base : DOCKER_TARGET=base
 build/base :
-	@(${DOCKER_BUILD_BASE})
+	@($(DOCKER_BUILD_BASE))
 
 .PHONY : build/forwarder
 build/forwarder : workspace/assets
 build/forwarder : base/get-infra-agent
 build/forwarder : DOCKER_TARGET=forwarder
-build/forwarder : DOCKER_IMAGE_NAME=${FWD_IMAGE_NAME}
+build/forwarder : DOCKER_IMAGE_NAME=$(FWD_IMAGE_NAME)
 build/forwarder :
-	@(${DOCKER_BUILD_BASE})
+	@($(DOCKER_BUILD_BASE))
 
 .PHONY : build/k8s-events-forwarder
 build/k8s-events-forwarder : workspace/assets
 build/k8s-events-forwarder : base/get-infra-agent
 build/k8s-events-forwarder : DOCKER_TARGET=k8s-events-forwarder
 build/k8s-events-forwarder : DOCKER_IMAGE_NAME=${K8S_FWD_IMAGE_NAME}
 build/k8s-events-forwarder :
-	@(${DOCKER_BUILD_BASE})
+	@($(DOCKER_BUILD_BASE))
 
 # publish to docker hub
 .PHONY : publish/base
 publish/base : build/base
 publish/base :
-	@(docker push ${BASE_IMAGE_NAME}:${IMAGE_VERSION})
-	@(docker push ${BASE_IMAGE_NAME}:latest)
+	@printf '\n================================================================\n'
+	@printf 'Target: publish/base\n'
+	@printf 'TAG: $(BASE_IMAGE_NAME):$(IMAGE_VERSION)'
+	@printf '\n================================================================\n'
+	@(docker push $(BASE_IMAGE_NAME):$(IMAGE_VERSION))
+	@(docker push $(BASE_IMAGE_NAME):latest)
 
 .PHONY : lint
 lint : lint/Dockerfile
 
 .PHONY : lint/Dockerfile
 lint/Dockerfile :
-	@(docker run --rm -i hadolint/hadolint < ${DOCKERFILE})
+	@(docker run --rm -i hadolint/hadolint < $(DOCKERFILE))
 
 .PHONY : clean
 clean:
-	@(rm -rf ${PROJECT_WORKSPACE})
+	@(rm -rf $(PROJECT_WORKSPACE))
+
+buildTargetArch=$(lastword $(subst -, ,$1))
+baseBuildTargetArch=$(lastword $(subst /, ,$1))
+
+.PHONY : build/base-arm
+build/base-arm :
+	@($(MAKE) build/base/$(call buildTargetArch, $@))
+
+.PHONY : build/base-arm64
+build/base-arm64 :
+	@($(MAKE) build/base/$(call buildTargetArch, $@))
+
+.PHONY : build/base-amd64
+build/base-amd64 :
+	@($(MAKE) build/base/$(call buildTargetArch, $@))
+
+# Helper target to build base images for each arch
+.PHONY : build/base/%
+build/base/% : export DOCKER_ARCH=$(call baseBuildTargetArch, $@)
+build/base/% : export USE_BUILDX=true
+build/base/% : export DOCKER_TAG_LATEST=$(DOCKER_BUILD_TAG_PREFIX)-$(DOCKER_ARCH)
+build/base/% :
+	@printf '\n================================================================\n'
+	@printf 'Target: build/base-$(call baseBuildTargetArch, $@)\n'
+	@printf 'DOCKER_TAG_LATEST: $(DOCKER_TAG_LATEST)'
+	@printf '\n================================================================\n'
+	@$(MAKE) build/base
+
+.PHONY : build/forwarder-arm
+build/forwarder-arm :
+	@($(MAKE) build/forwarder/$(call buildTargetArch, $@))
```

Comment:
why we use variables if we already know what arch it is?

Reply:
It calls `build/forwarder/arm` here which is a target that catches all `build/forwarder/%` and has all the logic to set correct arguments for the build. I like having a phony target here as it means that when using auto complete I can see there is a target to build the `arm` forwarder. If I just use the wilcard target I'll never know the make target to build the `arm` forwarder (unless I read the README 😅).